modules/ding_campaign_plus/modules/ding_campaign_plus_facet/ding_campaign_plus_facet.module

```diff
@@ -0,0 +1,1094 @@
+    '#type' => 'textfield',
+    '#title' => t('Title'),
+    '#default_value' => '',
+  );
+
+  $form['ding_campaign_plus_facets']['description'] = array(
+    '#prefix' => '<div class="description">',
+    '#suffix' => '</div>',
+    '#markup' => t('All facets are explained at <a href="!url">!url</a>', array('!url' => 'https://opensearch.addi.dk/b3.5_4.5/?showCqlFile&repository=prod&cql=opensearch_cql.xml#facet')),
+  );
+
+  $form += array('#submit' => array());
+  $form['#submit'][] = 'ding_campaign_plus_facet_admin_settings_form_submit';
+}
+
+/**
+ * Submit handler for the administration form.
+ */
+function ding_campaign_plus_facet_admin_settings_form_submit($form, $form_state) {
+  $facets = $form_state['values']['ding_campaign_plus_facets']['facets'];
+  foreach ($facets as $id => $facet) {
+    if (empty($facet['facet'])) {
+      unset($facets[$id]);
+    }
+  }
+
+  variable_set('ding_campaign_plus_facets', $facets);
+}
+
+/**
+ * Menu callback for facet delete confirm.
+ *
+ * @param $form_id
+ *   ID of the form calling the delete action.
+ * @param $form_state
+ *   The state of the form.
+ * @param $facet_id
+ *   The ID of the facets to remove.
+ *
+ * @return mixed
+ *   Confirmation for.
+ */
+function ding_campaign_plus_facet_delete($form_id, $form_state, $facet_id) {
+  $form['facet_id'] = array(
+    '#type' => 'value',
+    '#value' => $facet_id,
+  );
+  return confirm_form($form, t("Are you sure you want to delete @facet?", array('@facet' => $facet_id)), 'admin/config/ding/campaigns_plus');
+}
+
+/**
+ * Submit handler for the delete facet confirmation form.
+ */
+function ding_campaign_plus_facet_delete_submit($form, &$form_state) {
+  $facets = variable_get('ding_campaign_plus_facets', array());
+  foreach ($facets as $key => $facet) {
+    if ($facet['facet'] == $form_state['values']['facet_id']) {
+      unset($facets[$key]);
+    }
+  }
+  variable_set('ding_campaign_plus_facets', $facets);
+  drupal_set_message(t('@facet deleted.', array('@facet' => $form_state['values']['facet_id'])));
+  $form_state['redirect'] = 'admin/config/ding/campaigns_plus';
+}
+
+/**
+ * Default theme function for facet table.
+ *
+ * Add draggable table layout to facet table.
+ *
+ * @param array $vars
+ *   The variables for this theme function (form element).
+ *
+ * @return string
+ *  The formatted table.
+ */
+function theme_ding_campaign_plus_facet_table_drag($vars) {
+  drupal_add_tabledrag('campaign-facets', 'order', 'sibling', 'facet-weight');
+
+  $rows = array();
+  foreach (element_children($vars['element']) as $id) {
+    $facet = $vars['element'][$id];
+    $row = array();
+    $row[] = drupal_render($facet['facet']);
+    $row[] = drupal_render($facet['title']);
+    $row[] = drupal_render($facet['action']);
+
+    $rows[] = array(
+      'data' => $row,
+      'class' => array('draggable'),
+    );
+  }
+
+  $header = array(t('Facet'), t('Title'), t('Operations'));
+  return theme('table', array(
```

Comment:
Avoid calling this function directly. It is preferable to replace direct calls to the theme() function with calls to drupal_render() by passing a render array with a #theme key to drupal_render(), which in turn calls theme().

Reply:
See previous comment about tabledrag.